tests/test_image.py

```diff
@@ -0,0 +1,136 @@
+import urllib.error
+import urllib.request
+from io import BytesIO
+from math import ceil
+from unittest import mock
+
+from docx.shared import Inches
+from PIL import Image
+
+from html2docx.image import USABLE_HEIGHT, USABLE_WIDTH, image_size, load_image
+
+from .utils import PROJECT_DIR, TEST_DIR
+
+broken_image = PROJECT_DIR / "html2docx" / "image-broken.png"
+broken_image_bytes = broken_image.read_bytes()
+DPI = 72
+
+
+class TestLoadImage:
+    def test_basic(self):
+        image_data = load_image("https://via.placeholder.com/1x1.png")
+        expected = TEST_DIR / "data" / "1x1.png"
+        assert image_data.getbuffer() == expected.read_bytes()
+
+    def test_non_image(self):
+        image_data = load_image("https://www.erezlife.com/")
+        assert image_data.getbuffer() == broken_image_bytes
+
+    def test_bad_url(self):
+        image_data = load_image("bad")
+        assert image_data.getbuffer() == broken_image_bytes
+
+    def test_transient_network_error_retries(self):
+        url = "https://transient.network.issue.com/image.png"
+        with mock.patch(
+            "html2docx.image.urllib.request.urlopen",
+            autospec=True,
+            side_effect=urllib.error.URLError(
+                reason="[Errno -2] Name or service not known"
+            ),
+        ) as url_mock:
+            image_data = load_image(url)
+        assert url_mock.call_args_list == [mock.call(url)] * 3
+        assert image_data.getbuffer() == broken_image_bytes
+
+    def test_404(self):
+        with mock.patch(
+            "html2docx.image.urllib.request.urlopen",
+            autospec=True,
+            side_effect=urllib.error.HTTPError(
+                url="mocked", code=404, msg="Not found", hdrs=[], fp=BytesIO()
+            ),
+        ) as url_mock:
+            image_data = load_image("mocked")
+        # Did not retry.
+        url_mock.assert_called_once()
+        assert image_data.getbuffer() == broken_image_bytes
+
+
+def inches_to_px(inches: int, dpi: int = DPI) -> int:
+    return ceil(inches / Inches(1) * dpi)
+
+
+def px_to_inches(px: int, dpi: int = DPI) -> int:
+    return ceil(px * Inches(1) / dpi)
+
+
+def generate_image(width: int, height: int, dpi=(DPI, DPI)) -> BytesIO:
+    data = BytesIO()
+    with Image.new("L", (width, height)) as image:
+        image.save(data, format="png", dpi=dpi)
+    return data
+
+
+class TestImageSize:
```

Comment:
Same here.

If it is about organization, WDYT about separate files? But I'm find with them all in one.

Reply:
It is about organization, I’m not strongly attached to this but wanted to group the tests and the class seems a good solution. Both test the `image` module, so I figured they could live together in `test_image.py`, but I’m open to alternative layouts.

Reply:
My preference would be to organize these by files and drop the class. Up to you.